Updated contribution guidelines and developer steps (#41)
temporalio/documentation-legacy: README.md

> # Temporal documentation repository

Reviewer: don't put repository, it's obviously a repository :)

> This repo contains Temporal's open source documentation.

Reviewer: Why open source?

You can view the published documentation at https://docs.temporal.io.

> This repo utilizes the [Docusaurus 2](https://v2.docusaurus.io/) framework, a modern static website generator, which allows you to build and view the documentation locally as you contribute.

Reviewer: Don't use the word allow. Allow is bad because its what a parent does to a child. Use words that empower like "enable" or "makes it possible" even a neutral "provides". 

```suggestion
This repo utilizes the [Docusaurus 2](https://v2.docusaurus.io/) framework, a modern static website generator, which makes it possible to build and view the documentation locally.
```

Reviewer: That being said, hot reloading isn't exactly a unique feature to docusaurus

## Contribution guidelines

> Contributions from the community are highly encouraged.

Reviewer: Make these two sentences on a single line.

To contribute please follow these steps.

### Step 1: Fork and clone

> To ensure that each change is properly peer reviewed, we request that you follow standard open source contribution processes.

Reviewer: Is there a link we can provide?

> This means that you should start by forking the project into your own GitHub account.

Reviewer: ```suggestion
Start by forking the project into your own GitHub account and then cloning that fork onto your local machine.
```

Remove other sentence.

Author: These are two separate steps. Since it is followed by a command, one could be confused and assume that the command does both.

In this case it is better to have two separate sentences to address each step in the process.

I will add a screen shot after the first sentence -

Then, clone your own fork of the repo onto your local machine.

```
$ git clone git@github.com:<your-account>/documentation-legacy.git
```

### Step 2: Add upstream remote

> Then add the original repository as a new remote to your local clone.

Reviewer: ```suggestion
Using git, add the original repository as a new remote to your local clone.
```

```
$ cd documentation-legacy
$ git remote add upstream git@github.com/temporal/documentation-legacy.git
```

### Step 3: Create a local branch

> When you are ready to start making changes, sync your local clone with the upstream project and your forked GitHub project.

Reviewer: This is a heavy sentence.

```
$ git checkout master
$ git pull upstream master && git push origin master
```

> Then, create a branch to work on.

Reviewer: ```suggestion
Create a branch to work on.
```

Reviewer: I think you could use "then" less especially since it's already structured into steps.

```

> $ git checkout -b your-branch

Reviewer: don't know if this is actually needed right? Why do we care if they work off master in their fork.

Reviewer: Also can you add languages to the snippets

ie "```bash

     ```"

Author: Creating a branch is good practice. It also allows someone to work on multiple things at once. It is a step that is recommended in all of the contribution guides I have read including https://github.com/firstcontributions/first-contributions

```

> Now you can make your changes. See the next section about building and viewing the site locally for those steps.

Reviewer: This sentence could be a lot more digestible.

### Step 4: Submit a pull request

Once you are ready to submit your changes, create a pull request.

```
git push -u origin your-branch
```

This will create the branch on your GitHub project fork.

**Note** that the -u flag only needs to be used the first time you create a pull request for this branch.

> Next, visit the project fork in your GitHub account and locate the new branch, and click "Compare and pull request".

Reviewer: At this point I would either include pictures or defer to a guide of contributing to another repo.

Author: Linking to another guide.

> Ensure the "base fork" points to the correct repository and branch and provide a clear title and explanation of the changes with links to any issues that are being fixed by the change.

Reviewer: Never used the term "base fork" until here. What is it?

Author: This is a term used when comparing a branch which has an upstream remote attached -
It only shows up during the comparison and right before you click Create pull request.
This guide has a screen shot. https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/
I will grab a screen shot the next time I create a branch and compare - and add it to this README.

Then click "Create pull request".

## Building and viewing the site locally

This website is built using [Docusaurus 2](https://v2.docusaurus.io/).

Check out their [installation requirements](https://v2.docusaurus.io/docs/installation#requirements) for tooling prerequisites.

> ### Step 1: Initialize yarn

Reviewer: Tbh I don't think we need to explain yarn. If people want to learn about yarn they can use google etc. Just provide a 1 sentence summary and the commands IMO.

Reviewer: Probably can just comment the commands themselves inline if you want to provide more context.

Author: Since you had to explain to me that yarn start is actually executed via a custom script, and because it is not in their CLI docs, then I don't want to assume the reader knows anything about standard yarn usage. I would prefer to keep this information in the README.

The first step is to initialize the use of Yarn to build the site locally.

From your terminal, change directory to the project root and run:

```

> $ yarn

Reviewer 2: Let's also update this line to be explicit what it does

```suggestion
$ yarn install
```

```

### Step 2: Build the website

The following command generates static content into the `build` directory which can be served using any static contents hosting service.

```

> $ yarn build

Reviewer: You should never need to use this right?

Author: removing yarn build

```

It also ensures that the site is building correctly without errors or misconfigurations.

### Step 3: View the site in localhost

You can view the site via localhost in your browser by running this command from inside the project:

```
$ yarn start
```

This command starts a local development server and opens up a browser window.

> Most changes to the site are reflected instantly without having to restart the server.

Reviewer: What changes aren't?

> You can run `yarn start` independently of `yarn build`.

Reviewer: Remove 'build' references entirely IMO

## When will my changes be reflected in docs.temporal.io?

> As soon as your pull request is merged, a new build will automatically kick off and deploy the changes to the [next](https://docs.temporal.io/docs/next/) version of the docs.

Reviewer: Might as well mention you get a preview on PR submission

> Temporal will accumulate changes in the next version and periodically release them to the "current" version of the docs.

Reviewer: Can you reword this to indicate that our team manually does this and it's not some automatic CRON (periodically is ambiguous basically)

> Documentation version schemas will reflect the Temporal API version that they are pinned to as well as the iteration of the documentation for that API version.

Reviewer: Maybe take a screenshot of where this info is displayed on the site. The moment I read schemas as a user I'm going to zone out.

Author: I will once we adopt that schema and are using it live. At present, it would be confusing because we aren't doing that yet.

The website is built using [Docusaurus 2](https://v2.docusaurus.io/).

Check out their [installation requirements](https://v2.docusaurus.io/docs/installation#requirements) for tooling prerequisites.

### Initialize yarn

The first step is to initialize the use of Yarn to build the site locally.

From your terminal, change directory to the project root and run:

```bash
$ yarn install
```

Read the [yarn CLI docs](https://classic.yarnpkg.com/en/docs/cli/) if you want to learn what yarn is doing.

### View the site locally

You can view the site via localhost in your browser by running this command from inside the project:

```bash
$ yarn start
```

This command starts a local development server and opens up a browser window.

This will fail if there are source code errors or references to relative paths that do not exist.

Most changes to the site are reflected instantly without having to restart the server. Installing a new plugin will require you to restart.

## When will my changes be reflected in docs.temporal.io?

You will have a chance to preview the changes in your pull request. Click "Details" next to the Netlify deploy-preview check.

> ![Netlify build preview](static/img/readme/netlifypreview.png)

Reviewer: It seems nit picky but the image has a link in the picture which actually confused me enough that I tried to click it. Can you just take this pic without the link at the end.

As soon as your pull request is merged, a new build will automatically kick off and deploy the changes to the [next](https://docs.temporal.io/docs/next/) version of the docs.

Temporal documentation will accumulate changes in the "next" version and be periodically released by our team to the "current" version of the docs.

Documentation version schemas will reflect the Temporal API version that they are pinned to as well as the iteration of the documentation for that API version.

> Version schema: `{temporal api version}-docs-{docs version}`

Reviewer: Can you make these two colon items bold

**key**: value

# Temporal documentation

This repository contains Temporal's documentation.

The source of this documentation is available to the public for individual and commercial use.

You can view the published documentation at https://docs.temporal.io.

This documentation utilizes the [Docusaurus 2](https://v2.docusaurus.io/) framework, a modern static website generator, which makes it possible to build and view the documentation locally.

## Contribution guidelines

Contributions from the community are highly encouraged. To contribute please follow these steps.

### Fork and clone

To ensure that each change is relevant and properly peer reviewed, we request that you adhere to open source contribution best practices. GitHub's [first-contributions repo README](https://github.com/firstcontributions/first-contributions) offers a decent overview.

Start by forking the project into your own GitHub account.

> ![Fork repository button](static/img/readme/forkrepo.png)

Author: See if this improves. I increased the zoom in the browser and took the screen shot on the mac screen vs my 1080 x 720 screen.

> Using git, clone your fork onto your local machine.

Author: I like your suggestion to reference git. No harm in being clear.

Then click "Create pull request".

You can refer to [Rob Allen's DevNotes](https://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/) for details on this type of contribution process.

## Building and viewing the site locally

The website is built using [Docusaurus 2](https://v2.docusaurus.io/).

Check out their [installation requirements](https://v2.docusaurus.io/docs/installation#requirements) for tooling prerequisites.

### Initialize yarn

The first step is to initialize the use of Yarn to build the site locally.

From your terminal, change directory to the project root and run:

```bash
$ yarn install
```

> Read the [yarn CLI docs](https://classic.yarnpkg.com/en/docs/cli/) to learn what yarn is doing.

Author: Adopted the "if you want" part but I prefer placing the conjunction after the statement subject.
